dali/python/nvidia/dali/external_source.py

```diff
@@ -155,81 +155,98 @@ class ExternalSource():
-    explicitly.
-
-    Special values:
-      *  0 - use default CUDA stream
-      * -1 - use DALI's internal stream
-
-    If internal stream is used, the call to ``feed_input`` will block until the copy to internal
-    buffer is complete, since there's no way to synchronize with this stream to prevent
-    overwriting the array with new data in another stream.
+    If provided, sets the layout of the data.
+
+    When ``num_outputs > 1``, the layout can be a list that contains a distinct layout for each
+    output. If the list has fewer elements than the ``num_outputs`` value, only the first
+    outputs have the layout set, the rest of the outputs are cleared.
+
+`cuda_stream` : optional, ``cudaStream_t`` or an object convertible to ``cudaStream_t``, such as
+    ``cupy.cuda.Stream`` and ``torch.cuda.Stream``
+    The CUDA stream is going to be used to copy data to a GPU or from a GPU source.
+
+    If this value not set, a best-effort will be taken to maintain correctness. For example,
+    if the data is provided as a tensor/array from a recognized library such as CuPyor PyTorch,
+    the library’s current stream is used. Although this process works in typical scenarios,
+    with advanced use cases, and code that uses unsupported libraries, you might need to
+    explicitly supply the stream handle.
+
+    Here are the special values:
+      *  0 - Use the default CUDA stream
+      * -1 - Use DALI's internal stream
+
+    Since you cannot synchronize with this stream to prevent overwriting the array with new data
+    in another stream, if an internal stream is used, the call to ``feed_input`` will not complete
+    until the copy to the internal buffer is complete.
 
 `use_copy_kernel` : optional, `bool`
-    If set to True, DALI will use a CUDA kernel to feed the data (only applicable when copying
-    data to/from GPU memory) instead of cudaMemcpyAsync (default).
+    If set to True, DALI will use a CUDA kernel to feed the data instead of cudaMemcpyAsync (default).
+
+    .. note::
+        This is applicable only when copying data to and from GPU memory.
+
+`blocking` : optional,
+   Determines whether the external source should not complete until data is available or just fail
+   when the is not available.
 
-`blocking` : optional, Whether external source should block until data is available or just
-    fail when it is not
+`no_copy` : Determines whether DALI should copy the buffer when feed_input is called.
 
-`no_copy` : Whether DALI should copy the buffer when feed_input is called
-    If True, DALI passes the user memory directly to the Pipeline, instead of copying.
-    It is the user's responsibility to keep the buffer alive and unmodified
-    until it is consumed by the pipeline.
+    If set to True, DALI passes the user memory directly to the pipeline, instead of copying the
+    memory. It is your responsibility to keep the buffer alive and unmodified until it is
+    consumed by the pipeline.
```

Comment:
```suggestion
    If set to True, DALI passes the user memory directly to the pipeline, instead of copying it.
    It is your responsibility to keep the buffer alive and unmodified until it is
    consumed by the pipeline.
```

Reply:
Done